libgalois/include/katana/analytics/ClusteringImplementationBase.h

```diff
@@ -0,0 +1,680 @@
+      katana::PropertyGraph* pfg_from, katana::PropertyGraph* pfg_to,
+      const std::string& edge_property_name) {
+    const katana::GraphTopology& topology_from = pfg_from->topology();
+    const katana::GraphTopology& topology_to = pfg_to->topology();
+
+    KATANA_ASSERT(
+        (topology_from.num_nodes() == topology_to.num_nodes()),
+        "num nodes of both property graph must be same");
+
+    auto out_indices_view_from_result =
+        katana::ConstructPropertyView<katana::UInt64Property>(
+            topology_from.out_indices.get());
+    if (!out_indices_view_from_result) {
+      return out_indices_view_from_result.error();
+    }
+    auto out_indices_view_from =
+        std::move(out_indices_view_from_result.value());
+
+    auto out_dests_view_from_result =
+        katana::ConstructPropertyView<katana::UInt32Property>(
+            topology_from.out_dests.get());
+    if (!out_dests_view_from_result) {
+      return out_dests_view_from_result.error();
+    }
+    auto out_dests_view_from = std::move(out_dests_view_from_result.value());
+
+    auto out_indices_view_to_result =
+        katana::ConstructPropertyView<katana::UInt64Property>(
+            topology_to.out_indices.get());
+    if (!out_indices_view_to_result) {
+      return out_indices_view_to_result.error();
+    }
+    auto out_indices_view_to = std::move(out_indices_view_to_result.value());
+
+    auto out_dests_view_to_result =
+        katana::ConstructPropertyView<katana::UInt32Property>(
+            topology_to.out_dests.get());
+    if (!out_dests_view_to_result) {
+      return out_dests_view_to_result.error();
+    }
+    auto out_dests_view_to = std::move(out_dests_view_to_result.value());
+
+    /* First pass to find the number of edges */
+    katana::do_all(
+        katana::iterate((uint64_t)0, topology_from.num_nodes()),
+        [&](uint64_t n) { out_indices_view_to[n] = out_indices_view_from[n]; });
+    katana::do_all(
+        katana::iterate((uint64_t)0, topology_from.num_edges()),
+        [&](uint64_t e) { out_dests_view_to[e] = out_dests_view_from[e]; });
+
+    /* Copy edge properties */
+    using ArrowType = typename arrow::CTypeTraits<EdgeTy>::ArrowType;
+    auto edge_property_result =
+        pfg_from->GetEdgePropertyTyped<EdgeTy>(edge_property_name);
+    if (!edge_property_result) {
+      return edge_property_result.error();
+    }
+    auto edge_property = edge_property_result.value();
+    std::vector<std::shared_ptr<arrow::Field>> fields;
+    std::vector<std::shared_ptr<arrow::Array>> columns;
+    fields.emplace_back(
+        arrow::field((edge_property_name), std::make_shared<ArrowType>()));
+    columns.emplace_back(edge_property);
+    auto edge_data_table = arrow::Table::Make(arrow::schema(fields), columns);
+    if (auto r = pfg_to->AddEdgeProperties(edge_data_table); !r) {
+      return r;
+    }
+    return katana::ResultSuccess();
+  }
+
+  /*
+ * Graph construction routines to make
+ * coarser graphs.
+ */
```

Comment:
Either improve or delete these utterly useless comments that just restate the function names.

Reply:
Added more meaningful comments :).